### ISSUE TITLE
0.5.3 release causes constant RTM reconnects

### ISSUE THREAD

**Issue by jayjanssen** [howdyai/botkit#806]
I can confirm that 0.5.2 doesn't have this behavior. 

```
bot    │ notice: [botkit] ** BOT ID: perconabottest ...attempting to connect to RTM!
bot    │ notice: [botkit] RTM websocket opened
bot    │ info: [slack] slack RTM connection durationMs=1504
...
bot    │ notice: [botkit] RTM close event: 1006 :
bot    │ error: [botkit] Abnormal websocket close event, attempting to reconnect
...
bot    │ notice: [botkit] ** BOT ID: bot ...reconnect attempt #1 of 3 being made after 1000ms
bot    │ notice: [botkit] ** BOT ID: bot ...attempting to connect to RTM!
bot    │ notice: [botkit] RTM websocket opened
...
bot    │ notice: [botkit] RTM close event: 1006 :
bot    │ error: [botkit] Abnormal websocket close event, attempting to reconnect
bot    │ notice: [botkit] ** BOT ID: bot ...reconnect attempt #1 of 3 being made after 1000ms
bot    │ notice: [botkit] ** BOT ID: bot ...attempting to connect to RTM!
bot    │ notice: [botkit] RTM websocket opened
...
```

This will keep going on.

**Comment by dasblitz** [comment]
Same here

**Comment by derekolson** [comment]
Was just about to post this same issue.

**Comment by peterswimm** [comment]
What happens if you set debug to true? Do you get any more information?

**Comment by benbrown** [comment]
This may be due to an update in the websocket library we use.

I am investigating!

**Comment by jayjanssen** [comment]
Trying, the 5MB of JSON data bot kit prints to debug on RTM connect isn't helping...

```
bot    │ notice: [botkit] RTM websocket opened
bot    │ debug: [botkit] RECEIVED MESSAGE
bot    │ debug: [botkit] CUSTOM FIND CONVO undefined
bot    │ debug: [botkit] DEFAULT SLACK MSG RECEIVED RESPONDER
bot    │ debug: [botkit] RECEIVED MESSAGE
bot    │ debug: [botkit] CUSTOM FIND CONVO { SLACK USER PROFILE JSON OF ONE MY TEAM MEMBERS }
bot    │ debug: [botkit] DEFAULT SLACK MSG RECEIVED RESPONDER
bot    │ debug: [botkit] RECEIVED MESSAGE
bot    │ debug: [botkit] CUSTOM FIND CONVO undefined
bot    │ debug: [botkit] DEFAULT SLACK MSG RECEIVED RESPONDER
bot    │ debug: [botkit] RECEIVED MESSAGE
bot    │ debug: [botkit] CUSTOM FIND CONVO U0AF9HFF1
bot    │ debug: [botkit] DEFAULT SLACK MSG RECEIVED RESPONDER
.... LOTS OF JSON ....
bot    │ notice: [botkit] RTM close event: 1006 :
bot    │ error: [botkit] Abnormal websocket close event, attempting to reconnect
bot    │ notice: [botkit] ** BOT ID: bot ...reconnect attempt #1 of 3 being made after 1000ms
bot    │ debug: [botkit] rtm.start no_unreads=true, simple_latest=true, token=MY TOKEN
bot    │ info: [botkit] ** API CALL: https://slack.com/api/rtm.start



bot    │ notice: [botkit] RTM close event: 1006 :
bot    │ error: [botkit] Abnormal websocket close event, attempting to reconnect
bot    │ notice: [botkit] ** BOT ID: bot ...reconnect attempt #1 of 3 being made after 1000ms
bot    │ info: [botkit] ** API CALL: https://slack.com/api/rtm.start
bot    │ debug: [botkit] rtm.start no_unreads=true, simple_latest=true, token=MY TOKEN
```

**Comment by benbrown** [comment]
@jayjanssen if you downgrade the `ws` dependency back to 1.1.1, it should work.

not sure why this is breaking yet, but it appears to be related to that dependency.

**Comment by jayjanssen** [comment]
it's not urgent enough for me to build my own custom version.  I'm happy to test any new releases you come up with though.

**Comment by peterswimm** [comment]
With the help of Slack team, we think we have a fix for this, coming soon!

https://github.com/howdyai/botkit/pull/808

**Comment by anurag-saran** [comment]
I am seeing this issue with  "botkit": "^0.5.1" and  "botkit": "^0.5.0" versions too.

**Comment by anurag-saran** [comment]
By when are we expecting to get a fix ?

**Comment by jayjanssen** [comment]
@anurag-saran  I would expect "^0.5.1" and "^0.5.0" to upgrade to the latest in the 0.5.3 series.  Try just "0.5.2" without the ^

**Comment by anurag-saran** [comment]
Perfect this fixed the issue. Thanks a ton.

**Comment by benbrown** [comment]
This issue has been fixed in v0.5.4, now available on npm!
